### DISCUSSION TITLE
tool/aliyundrive/request

### DISCUSSION THREAD

**Discussion by giscus** [AlistGo/docs#164] [Announcements]
# tool/aliyundrive/request

import Request from 

https://alist.nn.ci/tool/aliyundrive/request.html

**Comment by CoffeeQaQ** [comment]
It's good!!!!!

**Comment by JivianJCL** [comment]
Failed init storage: failed to refresh token: The input parameter refresh_token is not valid.

**Reply by seaforest99** [reply]
me too

**Reply by hiwens** [reply]
metoo

**Reply by ViaVxxx** [reply]
Same

**Reply by AppleCZW** [reply]
metoo

**Reply by chuhaos** [reply]
the same

**Reply by agooddream9** [reply]
how to solve this problem?

**Reply by cjs520** [reply]
update your alist

**Reply by qherb** [reply]
the same

**Reply by DamienMM** [reply]
me too！！！！！！！！！！！！！！！！！！！！！！！

**Reply by arronnjxj** [reply]
The same，My  alist is v3.31.0.

**Comment by FopFat** [comment]
cool！

**Comment by dapengfpf** [comment]
eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJzdWIiOiIzNzZhMjJjNWNkMTc0ZmQxYjE1OGI4Njc4ZmU3ZDJhYSIsImF1ZCI6Ijc2OTE3Y2NjY2Q0NDQxYzM5NDU3YTA0ZjYwODRmYjJmIiwiZXhwIjoxNjg1MTU5NDAwLCJpYXQiOjE2NzczODM0MDB9.rqT1tsVqO9a_PZCpG7qQSb0QfcOm3wF1ywA2TtBgQNJ-WONZsGOsDqigbRL2qRWvjPRhVo41RpaA4ijrjRIYnw

**Comment by qwertyuiop6** [comment]
Failed init storage: failed to refresh token: The input parameter refresh_token is not valid.

**Comment by 1592363624** [comment]
状态: 

failed to refresh token: client_id 不能为空

客户端ID不能为空吗

是账号密码？

**Reply by anwen-anyi** [reply]
猜测你的 **`Oauth令牌链接`** 选项没写(默认是写好了的帮你
```html
https://api.nn.ci/alist/ali_open/token
```

**Reply by 1592363624** [reply]
> 猜测你的 **`Oauth令牌链接`** 选项没写(默认是写好了的帮你
> 
> ```
> https://api.nn.ci/alist/ali_open/token
> ```

被你猜对了=.=

**Comment by FopFat** [comment]
是B站里我关注的那个安稳大吗？

发自我的iPhone

**Comment by yuanpenghe** [comment]
eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJzdWIiOiIzNzZhMjJjNWNkMTc0ZmQxYjE1OGI4Njc4ZmU3ZDJhYSIsImF1ZCI6Ijc2OTE3Y2NjY2Q0NDQxYzM5NDU3YTA0ZjYwODRmYjJmIiwiZXhwIjoxNjg1MTU5NDAwLCJpYXQiOjE2NzczODM0MDB9.rqT1tsVqO9a_PZCpG7qQSb0QfcOm3wF1ywA2TtBgQNJ-WONZsGOsDqigbRL2qRWvjPRhVo41RpaA4ijrjRIYnw

**Reply by woshimaniubi8** [reply]
6

**Comment by 13280868508** [comment]
https://www.aliyundrive.com/drive

**Comment by ViaVxxx** [comment]
每天都要重新获取，不然就会失效

**Reply by wshaow** [reply]
我也是，这个有什么办法解决吗

**Comment by nimade2333** [comment]
这啥
failed get objs: failed to list objs: Post "https://open.aliyundrive.com/adrive/v1.0/openFile/list": dial tcp: lookup open.aliyundrive.com on [**1]:53: read udp [**1]:43159->[**1]:53: read: connection refused

**Reply by aop870325** [reply]
我也遇到这个问题了

**Reply by xfol** [reply]
我也是

**Comment by XavierProduct** [comment]
failed get objs: failed to list objs: Post "https://api.nn.ci/alist/ali_open/token": read tcp **********:40322->********08:443: read: connection reset by peer
经常出现这个

**Reply by aop870325** [reply]
解决了吗

**Comment by GlodenBoy** [comment]
Post "https://open.aliyundrive.com/oauth/access_token": net/http: TLS handshake timeout

**Reply by duanxinustb** [reply]
me too

**Reply by wujiyu305** [reply]
same here

**Reply by potatocat857** [reply]
metoooo

**Reply by GlodenBoy** [reply]
官网看到Api 有变动，估计需要alist更随跟新

**Comment by nongmin635000** [comment]
Error
Post "https://open.aliyundrive.com/oauth/authorize/qrcode": net/http: TLS handshake timeout

点击登录就提示这个，是怎么回事喃

**Comment by wanglei-github** [comment]
Error
Post "https://open.aliyundrive.com/oauth/authorize/qrcode": net/http: TLS handshake timeout
老板们，帮忙看下

**Comment by lovesnsfi** [comment]
there is a error,like this
Post "https://open.aliyundrive.com/oauth/access_token": net/http: TLS handshake timeout

**Comment by DamienMM** [comment]
http: TLS handshake timeout！！！

**Comment by xiyuyeyun** [comment]
这2天经常打不开无法访问

**Comment by xiyuyeyun** [comment]
太麻烦了，三天两头的



发自我的iPhone


------------------ Original ------------------
From: Ninety ***@***.***&gt;
Date: Sun,Mar 19,2023 7:24 PM
To: alist-org/docs ***@***.***&gt;
Cc: xiyuyeyun ***@***.***&gt;, Comment ***@***.***&gt;
Subject: Re: [alist-org/docs] tool/aliyundrive/request (Discussion #164)





 
官网看到Api 有变动，估计需要alist更随跟新

**Comment by lovesnsfi** [comment]
估计是官网接口更新太频繁了发自我的 iPhone在 2023年3月19日，19:38，xiyuyeyun ***@***.***> 写道：﻿
太麻烦了，三天两头的







发自我的iPhone





------------------ Original ------------------

From: Ninety ***@***.***&gt;

Date: Sun,Mar 19,2023 7:24 PM

To: alist-org/docs ***@***.***&gt;

Cc: xiyuyeyun ***@***.***&gt;, Comment ***@***.***&gt;

Subject: Re: [alist-org/docs] tool/aliyundrive/request (Discussion #164)











 

官网看到Api 有变动，估计需要alist更随跟新

**Comment by hschr** [comment]
用不了了

**Comment by DarinJi** [comment]
too many request 怎么办 还能用吗

**Reply by GlodenBoy** [reply]
已恢复

**Comment by zhstzzy** [comment]
too many request 怎么办

**Reply by wshidaijingshen** [reply]
等待。
官方文档有写：

```
同一 IP 在 60 分钟内请求 10 次，会出现 TooManyRequests 🚫避免滥用，请勿滥用🚫
例如在 保存/编辑 的时候算一次请求，查看文件看视频下载不算。
上述的分钟和次数是动态的喔~
```



靠，我还以为 是 阿里出问题了，原来是alist 自己的限制，
可惜了，把之前配置好的别人泄露的 ali盘都删除了，才发现，为时已晚。

**Comment by yja244150061** [comment]
怎么获取不了啊。一直在转转转

**Reply by FopFat** [reply]
是的.一样的问题

**Comment by maotui** [comment]
ailed to initialize storage, but storage was created:Failed to initialize storage: Failed to refresh token:Application does not exist

**Comment by FopFat** [comment]
点击“Scan QrCode”出现Error。然后选择登陆,也刷不出来token.求解
2023年4月8日23:13:59

**Comment by Adrian19970116** [comment]
eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJzdWIiOiI0ZGFjZjU4MjA0M2E0NmNjYjlhMWFkOWJlNGIyMWVkOCIsImF1ZCI6Ijc2OTE3Y2NjY2Q0NDQxYzM5NDU3YTA0ZjYwODRmYjJmIiwiZXhwIjoxNjkwMjA5NTQwLCJpYXQiOjE2ODI0MzM1NDB9.arcU4xS7OgIXY6_Kv5r2y6GI3KC038y-gKwlPiiRVBs5nM1h1dPLOafeQckBrXwmjjnzaQl47Y92Z-apCbkgJg

**Comment by ChaXxl** [comment]
What the hell？
~~~shell
Error
Too Many Requests
~~~

**Comment by evianincrush** [comment]
您好，我明明已经按照以往的提示，获取token，但在保存时出现这个提示“TokenVerifyFailed:The token was expected to have 3 parts, but got 1.”请问这是什么原因呀?该如何解决呢?

**Comment by Lancomebro** [comment]
failed get objs: failed to list objs: AccessTokenExpired:token expired 。重新获取后没过几个小时token又过期了。

**Comment by Febare** [comment]
Failed init storage: TokenVerifyFailed:The token was expected to have 3 parts, but got 1.

**Comment by 1416261278** [comment]
Failed init storage: TokenVerifyFailed:The token was expected to have 3 parts, but got 1.

**Comment by James4518** [comment]
refresh_token is not valid

**Comment by coinca** [comment]
每次更新完令牌，过几个小时要么JWT expired，要么toekn expired。
改config,json里的jwt过期时间、改设置里的token过期时间都没用。
挣扎了几个月，最新版还是这毛病，改用webdav去了。垃圾玩意。

**Reply by dustin126** [reply]
是的，每天晚上想看infuse了，都要重新获取令牌真的受不了

**Comment by WYLK0912** [comment]
# token获取失败，也获取不了二维码
早上发现阿里云挂载失败后，准备重新获取token，发现登录后加载不出token，qrcode也无法加载出来

**Reply by JayHome137** [reply]
oauth令牌链接 更换成 https://api-cf.nn.ci/alist/ali_open/token 应该获取令牌的网站出问题了 上面获取令牌的网站还有2个备用的 可以自己试试看

**Comment by GoofyBen82** [comment]
Post "https://api.nn.ci/alist/ali_open/token": context deadline exceeded (Client.Timeout exceeded while awaiting headers)，怎么破？

**Reply by JayHome137** [reply]
oauth令牌链接 更换成 https://api-cf.nn.ci/alist/ali_open/token 应该获取令牌的网站出问题了 上面获取令牌的网站还有2个备用的 可以自己试试看

**Comment by JayHome137** [comment]
oauth令牌链接 更换成 https://api-cf.nn.ci/alist/ali_open/token 应该获取令牌的网站出问题了 上面获取令牌的网站还有2个备用的 可以自己试试看

**Comment by yull1860outlook** [comment]
tried https://api.xhofe.top/alist/ali_open/token  or api-cf.nn.ci both not work . looks the token not valid , jwt.io show invalid signature

**Comment by hoadi1230** [comment]
状态: 

TokenVerifyFailed:The token was expected to have 3 parts, but got 1.

why？？？

**Comment by KolinLoa** [comment]
# token verify failed
token·是没问题的，但是填入后提示token is 3 parts ， but got 1

**Reply by hoadi1230** [reply]
我刚开始也提示这个，但是过了一阵子重新试了下，突然就好了，我是按照这个帖子提供的方法，不知道问题出在哪里了。

**Reply by KolinLoa** [reply]
我试了好几次，一直不行，我都怀疑是不是手机输入法有问题

**Reply by hoadi1230** [reply]
有空就多试几次，可能就好了，我就是反复试几次，https://api-cf.nn.ci/alist/ali_open/token
https://api.xhofe.top/alist/ali_open/token 令牌链接换下这两个试试。

**Comment by JayHome137** [comment]
But I am worked and work till now

Best Regards
Jay

在 2023年6月13日，22:40，yull1860outlook ***@***.***> 写道：

﻿

tried https://api.xhofe.top/alist/ali_open/token or api-cf.nn.ci both not work . looks the token not valid , jwt.io show invalid signature

**Comment by leekakoo** [comment]
Failed init storage: TokenVerifyFailed:The token was expected to have 3 parts, but got 1.

**Comment by bigbigking** [comment]
这个错误怎么弄？
Failed init storage: TokenVerifyFailed:The token was expected to have 3 parts, but got 1.

**Reply by bigbigking** [reply]
解决了，更新到最新版本

**Comment by kakolukia** [comment]
Thanks for this amazing tool!

**Comment by ghostry** [comment]
go to login has error https://dev.g.alicdn.com/aliyun-drive-fe/aliyun-drive/4.4.0-auth/auth/bundle.js timeout

**Comment by suifenggyy** [comment]
Aliyundrive Refresh Token Callback 一直转圈，刷新不出来

**Comment by wang2580liang** [comment]
docs/zh/guide/drivers/aliyundrive_open.md

**Comment by Sh4p3s** [comment]
Failed init storage: failed to refresh token: app not exists , 
What can I do to fix it，pls tell me.

**Comment by shentong0722** [comment]
用不了啊，能列出列表，打开文件报错，这是啥呀完全看不懂：
failed link: failed get link: RpcException:Failed to invoke the method isUserConsumeLimit in the service com.alibaba.ic.adrive.contract.dubbo.UserCapacityDubboService. Tried 1 times of the providers [*********35:20880] (1/120) from the registry mse-bd5d2222-zk.mse.aliyuncs.com:2181 on the consumer *********56 using the dubbo version 2.6.12. Last error is: Invoke remote method timeout. method: isUserConsumeLimit, provider: dubbo://*********35:20880/com.alibaba.ic.adrive.contract.dubbo.UserCapacityDubboService?__micro.service.app.id__=c576f2de-4247-45e6-bab0-60346831cabc&__micro.service.env__=[{"desc":"edas-canary","priority":-99,"tag":"183cb207cb","type":"canary"}]&anyhost=true&application=adrive-open-biz&bean.name=ServiceBean:com.alibaba.ic.adrive.contract.dubbo.UserCapacityDubboService&check=false&default.check=false&default.retries=0&default.service.filter=-exception&default.timeout=3000&dubbo=2.0.2&generic=false&interface=com.alibaba.ic.adrive.contract.dubbo.UserCapacityDubboService&methods=isUserConsumeLimit&mse.appId=c576f2de-4247-45e6-bab0-60346831cabc&mse_aliyun_com__zone=cn-beijing-h&opensergo.io/canary=&pid=96&register.ip=*********56&remote.timestamp=1702981303285&revision=********0231213-1-SNAPSHOT&side=consumer&timeout=3000&timestamp=1702619004788, cause: Waiting server-side response timeout by scan timer. start time: 2023-12-22 11:14:39.484, end time: 2023-12-22 11:14:42.487, client elapsed: 0 ms, server elapsed: 3003 ms, timeout: 3000 ms, request: Request [id=66810435, version=2.0.2, twoway=true, event=false, broken=false, data=RpcInvocation [methodName=isUserConsumeLimit, parameterTypes=[class java.lang.String], arguments=[b3334178cdca4a7c8c9480ba4c44e308], attachments={__microservice_match_result__=[], EagleEye-pSpanID=8675862433707920330, EagleEye-pAppName=95ecab6c-994e-47f9-b6e3-195b9ff3a36b, EagleEye-IP=*********56, interface=com.alibaba.ic.adrive.contract.dubbo.UserCapacityDubboService, version=0.0.0, timeout=3000, EagleEye-UserData=__microservice_log_cur_span_id__=8675862433707920330&__microservice_match_result__=[]&mse_alf=0&_arms.trace.sc=s6, path=com.alibaba.ic.adrive.contract.dubbo.UserCapacityDubboService, EagleEye-RpcID=0.7, EagleEye-ROOT-APP=95ecab6c-994e-47f9-b6e3-195b9ff3a36b, dubboApplication=adrive-open-biz, EagleEye-TraceID=ea1a012d9c17032148791137332d0060, __micro.service.app.id__=95ecab6c-994e-47f9-b6e3-195b9ff3a36b, EagleEye-SpanID=-3873502083589789294, EagleEye-pRpc=/v1.0/openFile/getDownloadUrl}, attributes={serialization_id=2}]], channel: /*********56:45816 -> /*********35:20880

**Comment by woshimaniubi8** [comment]
This very NB

**Comment by ENDDREAM-ZM** [comment]
Failed load storage: failed init storage: TokenVerifyFailed:The token was expected to have 3 parts, but got 1.
怎么解决啊🤕

**Reply by ENDDREAM-ZM** [reply]
阿里盘 版本3.15.0

**Comment by swguxing111** [comment]
gooooood

**Comment by xingxinag** [comment]
https://api.xhofe.top 
这个接口老是报错
好像用不了了

**Comment by hox521** [comment]
dfdfdf

**Comment by zhychen1173** [comment]
我用 https://alist.nn.ci/tool/aliyundrive/request.html 使用自己的client_id 和 client_secret ，然后Scan QRcode ，手机app 授权之后，点击 I have scan，跳转页面报错，报：InvalidClientId: invalid client_id
InvalidClientId: invalid client_id

请教应该如何解决？谢谢。

**Reply by zhychen1173** [reply]
手机阿里云盘app点击授权之后，到阿里云盘app的授权管理里去看，是已经授权成功了的，但就是这个页面跳转获取token报错，报 InvalidClientId: invalid client_id 。

**Reply by decade20092009** [reply]
请问解决了吗，我也遇到一样的问题

**Comment by hbjsmzy** [comment]
eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJzdWIiOiI0NTVjYjc0MDFhZjQ0OGQxYWY4ZGU4MGQ1ZWIwNTIxNCIsImF1ZCI6Ijc2OTE3Y2NjY2Q0NDQxYzM5NDU3YTA0ZjYwODRmYjJmIiwiZXhwIjoxNzI2NDk4NjM5LCJpYXQiOjE3MTg3MjI2MzksImp0aSI6ImFmMTBkZWNhOWRkNTQyZDc5ZWI3MWIyNjNhZmQwYzQxIn0.uXWiZvrHnLmRhiZNWoGn39VfMB869YQS9Ce6y62RYK9C2g_tgEXhQ8XFE3XH1j4VhKQAnW36yEHI77IZPoIAmw

**Comment by watergaoer** [comment]
https://api-cf.nn.ci/alist/ali_open/token

**Comment by watergaoer** [comment]
https://api.xhofe.top/alist/ali_open/token

**Comment by labadapt** [comment]
sdjklqAJBHKSWASJKamkSJIOALFDSUDUYAIUDi8awdu8sauidasuidasuida

**Comment by labadapt** [comment]
<sup>eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJzdWIiOiI0NTVjYjc0MDFhZjQ0OGQxYWY4ZGU4MGQ1ZWIwNTIxNCIsImF1ZCI6Ijc2OTE3Y2NjY2Q0NDQxYzM5NDU3YTA0ZjYwODRmYjJmIiwiZXhwIjoxNzI2NDk4NjM5LCJpYXQiOjE3MTg3MjI2MzksImp0aSI6ImFmMTBkZWNhOWRkNTQyZDc5ZWI3MWIyNjNhZmQwYzQxIn0.uXWiZvrHnLmRhiZNWoGn39VfMB869YQS9Ce6y62RYK9C2g_tgEXhQ8XFE3XH1j4VhKQAnW36yEHI77IZPoIAmw</sup><sub>eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJzdWIiOiI0NTVjYjc0MDFhZjQ0OGQxYWY4ZGU4MGQ1ZWIwNTIxNCIsImF1ZCI6Ijc2OTE3Y2NjY2Q0NDQxYzM5NDU3YTA0ZjYwODRmYjJmIiwiZXhwIjoxNzI2NDk4NjM5LCJpYXQiOjE3MTg3MjI2MzksImp0aSI6ImFmMTBkZWNhOWRkNTQyZDc5ZWI3MWIyNjNhZmQwYzQxIn0.uXWiZvrHnLmRhiZNWoGn39VfMB869YQS9Ce6y62RYK9C2g_tgEXhQ8XFE3XH1j4VhKQAnW36yEHI77IZPoIAmw</sub>

**Reply by labadapt** [reply]
sdjklzxjka>KL:
ll

**Comment by xfol** [comment]
Post "https://openapi.alipan.com/adrive/v1.0/user/getDriveInfo": dial tcp: lookup openapi.alipan.com on 202.103.44.150:53: read udp 172.17.0.2:55652->202.103.44.150:53: i/o timeout
